corehq/form_processor/utils/general.py

```diff
@@ -26,7 +26,18 @@ def clear_local_domain_sql_backend_override(domain):
     _thread_local.use_sql_backend = use_sql_backend_dict
 
 
+def _short_circuit_sql_check(domain_object_or_name):
+    from corehq.apps.domain.models import Domain
+    domain_name = domain_object_or_name
+    if isinstance(domain_object_or_name, Domain):
+        domain_name = domain_object_or_name.name
+
+    return domain_name == 'icds-cas'
+
+
 def should_use_sql_backend(domain_object_or_name):
+    if _short_circuit_sql_check(domain_object_or_name):
```

Comment:
why didn't you use `ENTERPRISE_OPTIMIZATIONS`? https://github.com/dimagi/commcare-hq/blob/master/corehq/toggles.py#L1248